Funcky.Test/Extensions/StringExtensions/SplitLinesTest.cs

```diff
@@ -0,0 +1,47 @@
+using System.Linq;
+using Funcky.Extensions;
+using Xunit;
+
+namespace Funcky.Test.Extensions.StringExtensions
+{
+    public sealed class SplitLinesTest
+    {
+        [Fact]
+        public void SplitLinesOnAnEmptyStringReturnsAnEmptySequence()
+        {
+            Assert.Equal(Enumerable.Empty<string>(), string.Empty.SplitLines());
+        }
+
+        [Fact]
+        public void SplitLinesOnAStringWithoutANewLineCharacterReturnsTheString()
+        {
+            var text = "single line text";
+
+            Assert.Equal(Sequence.Return(text), text.SplitLines());
+        }
+
+        [Fact]
+        public void ASingleNewLineSplitsInASingleEmptyLine()
+        {
+            var text = "\n";
+
+            Assert.Equal(Sequence.Return(string.Empty), text.SplitLines());
+        }
+
+        [Fact]
+        public void IfANewLineIsAtTheEndOfTheLastLineNoEmptyLineIsAdded()
```

Comment:
I'd make this into a theory for each newline separator

Reply:
Addressed in 388b8bd2